<Type Name="ArithmeticException" FullName="System.ArithmeticException">
  <Metadata><Meta Name="ms.openlocfilehash" Value="d4091774aeb94d964164e19f7b72854106fd63b7" /><Meta Name="ms.sourcegitcommit" Value="c0c07dbd19cd7017243f9ac36915755f79bc8da6" /><Meta Name="ms.translationtype" Value="MT" /><Meta Name="ms.contentlocale" Value="zh-TW" /><Meta Name="ms.lasthandoff" Value="11/27/2018" /><Meta Name="ms.locfileid" Value="52362315" /></Metadata><TypeSignature Language="C#" Value="public class ArithmeticException : SystemException" />
  <TypeSignature Language="ILAsm" Value=".class public auto ansi serializable beforefieldinit ArithmeticException extends System.SystemException" />
  <TypeSignature Language="DocId" Value="T:System.ArithmeticException" />
  <TypeSignature Language="VB.NET" Value="Public Class ArithmeticException&#xA;Inherits SystemException" />
  <TypeSignature Language="C++ CLI" Value="public ref class ArithmeticException : SystemException" />
  <TypeSignature Language="F#" Value="type ArithmeticException = class&#xA;    inherit SystemException" />
  <AssemblyInfo>
    <AssemblyName>System.Runtime</AssemblyName>
    <AssemblyVersion>4.0.0.0</AssemblyVersion>
    <AssemblyVersion>4.0.10.0</AssemblyVersion>
    <AssemblyVersion>4.0.20.0</AssemblyVersion>
    <AssemblyVersion>4.1.0.0</AssemblyVersion>
    <AssemblyVersion>4.2.0.0</AssemblyVersion>
    <AssemblyVersion>4.2.1.0</AssemblyVersion>
  </AssemblyInfo>
  <AssemblyInfo>
    <AssemblyName>mscorlib</AssemblyName>
    <AssemblyVersion>1.0.5000.0</AssemblyVersion>
    <AssemblyVersion>2.0.0.0</AssemblyVersion>
    <AssemblyVersion>2.0.5.0</AssemblyVersion>
    <AssemblyVersion>4.0.0.0</AssemblyVersion>
  </AssemblyInfo>
  <AssemblyInfo>
    <AssemblyName>netstandard</AssemblyName>
    <AssemblyVersion>2.0.0.0</AssemblyVersion>
  </AssemblyInfo>
  <Base>
    <BaseTypeName>System.Exception</BaseTypeName>
    <BaseTypeName FrameworkAlternate="netframework-4.5">System.SystemException</BaseTypeName>
  </Base>
  <Interfaces />
  <Attributes>
    <Attribute FrameworkAlternate="netframework-2.0;netframework-3.0;netframework-3.5;netframework-4.0;netframework-4.5;netframework-4.5.1;netframework-4.5.2;netframework-4.6;netframework-4.6.1;netframework-4.6.2;netframework-4.7;netframework-4.7.1;netframework-4.7.2;xamarinandroid-7.1;xamarinios-10.8;xamarinmac-3.0;netframework-4.8">
      <AttributeName>System.Runtime.InteropServices.ComVisible(true)</AttributeName>
    </Attribute>
    <Attribute FrameworkAlternate="netframework-1.1;netframework-2.0;netframework-3.0;netframework-3.5;netframework-4.0;netframework-4.5;netframework-4.5.1;netframework-4.5.2;netframework-4.6;netframework-4.6.1;netframework-4.6.2;netframework-4.7;netframework-4.7.1;netframework-4.7.2;netframework-4.8;xamarinandroid-7.1;xamarinios-10.8;xamarinmac-3.0">
      <AttributeName>System.Serializable</AttributeName>
    </Attribute>
  </Attributes>
  <Docs>
    <summary><span data-ttu-id="4a52c-101">當算術、轉型或轉換作業發生錯誤時，所擲回的例外狀況。</span><span class="sxs-lookup"><span data-stu-id="4a52c-101">The exception that is thrown for errors in an arithmetic, casting, or conversion operation.</span></span></summary>
    <remarks>
      <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="4a52c-102"><xref:System.ArithmeticException> 是下列例外狀況的基底類別：</span><span class="sxs-lookup"><span data-stu-id="4a52c-102"><xref:System.ArithmeticException> is the base class for the following exceptions:</span></span>  
  
-   <span data-ttu-id="4a52c-103"><xref:System.DivideByZeroException>其中中擲回整數除法除數為 0。</span><span class="sxs-lookup"><span data-stu-id="4a52c-103"><xref:System.DivideByZeroException>, which is thrown in integer division when the divisor is 0.</span></span>  <span data-ttu-id="4a52c-104">例如，嘗試將 10 由 0 會擲回<xref:System.DivideByZeroException>例外狀況。</span><span class="sxs-lookup"><span data-stu-id="4a52c-104">For example, attempting to divide 10 by 0 throws a <xref:System.DivideByZeroException> exception.</span></span>  
  
-   <span data-ttu-id="4a52c-105"><xref:System.NotFiniteNumberException>其中時擲回一項作業上執行時，或傳回<xref:System.Double.NaN?displayProperty=nameWithType>， <xref:System.Double.NegativeInfinity?displayProperty=nameWithType>， <xref:System.Double.PositiveInfinity?displayProperty=nameWithType>， <xref:System.Single.NaN?displayProperty=nameWithType>， <xref:System.Single.NegativeInfinity?displayProperty=nameWithType>， <xref:System.Single.PositiveInfinity?displayProperty=nameWithType>，且使用的程式設計語言不支援這些值。</span><span class="sxs-lookup"><span data-stu-id="4a52c-105"><xref:System.NotFiniteNumberException>, which is thrown when an operation is performed on or returns <xref:System.Double.NaN?displayProperty=nameWithType>, <xref:System.Double.NegativeInfinity?displayProperty=nameWithType>, <xref:System.Double.PositiveInfinity?displayProperty=nameWithType>, <xref:System.Single.NaN?displayProperty=nameWithType>, <xref:System.Single.NegativeInfinity?displayProperty=nameWithType>, <xref:System.Single.PositiveInfinity?displayProperty=nameWithType>, and the programming language used does not support those values.</span></span>  
  
-   <span data-ttu-id="4a52c-106"><xref:System.OverflowException>其會擲回作業的結果超出界限的目標資料類型。</span><span class="sxs-lookup"><span data-stu-id="4a52c-106"><xref:System.OverflowException>, which is thrown when the result of an operation is outside the bounds of the target data type.</span></span> <span data-ttu-id="4a52c-107">也就是說，它是小於數字`MinValue`屬性或大於其`MaxValue`屬性。</span><span class="sxs-lookup"><span data-stu-id="4a52c-107">That is, it is less than a number's `MinValue` property or greater than its `MaxValue` property.</span></span>  <span data-ttu-id="4a52c-108">例如，嘗試指派 200 + 200<xref:System.Byte>值都會擲回<xref:System.OverflowException>例外狀況，因為大於 256，上限 400<xref:System.Byte>資料型別。</span><span class="sxs-lookup"><span data-stu-id="4a52c-108">For example, attempting to assign 200 + 200 to a <xref:System.Byte> value throws an <xref:System.OverflowException> exception, since 400 greater than 256, the upper bound of the <xref:System.Byte> data type.</span></span>  
  
 <span data-ttu-id="4a52c-109">您的程式碼不應該處理，或擲回這個例外狀況。</span><span class="sxs-lookup"><span data-stu-id="4a52c-109">Your code should not handle or throw this exception.</span></span> <span data-ttu-id="4a52c-110">相反地，您應該處理，或擲回其中一個衍生的類別，因為它更精確地指出錯誤的確實性質。</span><span class="sxs-lookup"><span data-stu-id="4a52c-110">Instead, you should either handle or throw one of its derived classes, since it more precisely indicates the exact nature of the error.</span></span>  
  
 <span data-ttu-id="4a52c-111">如需執行個體的初始屬性值的清單<xref:System.ArithmeticException>，請參閱<xref:System.ArithmeticException>建構函式。</span><span class="sxs-lookup"><span data-stu-id="4a52c-111">For a list of initial property values for an instance of <xref:System.ArithmeticException>, see the <xref:System.ArithmeticException> constructors.</span></span>  
  
 <span data-ttu-id="4a52c-112"><xref:System.ArithmeticException> 會使用 HRESULT COR_E_ARITHMETIC，其值 0x80070216。</span><span class="sxs-lookup"><span data-stu-id="4a52c-112"><xref:System.ArithmeticException> uses the HRESULT COR_E_ARITHMETIC, which has the value 0x80070216.</span></span>  
  
 ]]></format>
    </remarks>
    <altmember cref="T:System.Exception" />
    <related type="Article" href="~/docs/standard/exceptions/index.md"><span data-ttu-id="4a52c-113">在 .NET 中處理和擲回例外狀況</span><span class="sxs-lookup"><span data-stu-id="4a52c-113">Handling and throwing exceptions in .NET</span></span></related>
  </Docs>
  <Members>
    <MemberGroup MemberName=".ctor">
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.10.0</AssemblyVersion>
        <AssemblyVersion>4.0.20.0</AssemblyVersion>
        <AssemblyVersion>4.1.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Docs>
        <summary><span data-ttu-id="4a52c-114">初始化 <see cref="T:System.ArithmeticException" /> 類別的新執行個體。</span><span class="sxs-lookup"><span data-stu-id="4a52c-114">Initializes a new instance of the <see cref="T:System.ArithmeticException" /> class.</span></span></summary>
      </Docs>
    </MemberGroup>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public ArithmeticException ();" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor() cil managed" />
      <MemberSignature Language="DocId" Value="M:System.ArithmeticException.#ctor" />
      <MemberSignature Language="VB.NET" Value="Public Sub New ()" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; ArithmeticException();" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.10.0</AssemblyVersion>
        <AssemblyVersion>4.0.20.0</AssemblyVersion>
        <AssemblyVersion>4.1.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>1.0.5000.0</AssemblyVersion>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters />
      <Docs>
        <summary><span data-ttu-id="4a52c-115">初始化 <see cref="T:System.ArithmeticException" /> 類別的新執行個體。</span><span class="sxs-lookup"><span data-stu-id="4a52c-115">Initializes a new instance of the <see cref="T:System.ArithmeticException" /> class.</span></span></summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="4a52c-116">這個建構函式初始化<xref:System.Exception.Message%2A>屬性的新執行個體的系統提供的訊息，說明錯誤，例如 「 不允許算術運算。 」</span><span class="sxs-lookup"><span data-stu-id="4a52c-116">This constructor initializes the <xref:System.Exception.Message%2A> property of the new instance to a system-supplied message that describes the error, such as "The arithmetic operation is not allowed."</span></span> <span data-ttu-id="4a52c-117">此訊息會考量目前的系統文化特性。</span><span class="sxs-lookup"><span data-stu-id="4a52c-117">This message takes into account the current system culture.</span></span>  
  
 <span data-ttu-id="4a52c-118">下表顯示的執行個體的初始屬性值<xref:System.ArithmeticException>。</span><span class="sxs-lookup"><span data-stu-id="4a52c-118">The following table shows initial property values for an instance of <xref:System.ArithmeticException>.</span></span>  
  
|<span data-ttu-id="4a52c-119">屬性</span><span class="sxs-lookup"><span data-stu-id="4a52c-119">Property</span></span>|<span data-ttu-id="4a52c-120">值</span><span class="sxs-lookup"><span data-stu-id="4a52c-120">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="4a52c-121">`null`.</span><span class="sxs-lookup"><span data-stu-id="4a52c-121">`null`.</span></span>|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="4a52c-122">當地語系化的錯誤訊息字串。</span><span class="sxs-lookup"><span data-stu-id="4a52c-122">The localized error message string.</span></span>|  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public ArithmeticException (string message);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.ArithmeticException.#ctor(System.String)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (message As String)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; ArithmeticException(System::String ^ message);" />
      <MemberSignature Language="F#" Value="new ArithmeticException : string -&gt; ArithmeticException" Usage="new System.ArithmeticException message" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.10.0</AssemblyVersion>
        <AssemblyVersion>4.0.20.0</AssemblyVersion>
        <AssemblyVersion>4.1.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>1.0.5000.0</AssemblyVersion>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="message" Type="System.String" />
      </Parameters>
      <Docs>
        <param name="message"><span data-ttu-id="4a52c-123">描述錯誤的 <see cref="T:System.String" />。</span><span class="sxs-lookup"><span data-stu-id="4a52c-123">A <see cref="T:System.String" /> that describes the error.</span></span></param>
        <summary><span data-ttu-id="4a52c-124">使用指定的錯誤訊息，初始化 <see cref="T:System.ArithmeticException" /> 類別的新執行個體。</span><span class="sxs-lookup"><span data-stu-id="4a52c-124">Initializes a new instance of the <see cref="T:System.ArithmeticException" /> class with a specified error message.</span></span></summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="4a52c-125">`message` 參數的內容必須能讓人了解。</span><span class="sxs-lookup"><span data-stu-id="4a52c-125">The content of the `message` parameter is intended to be understood by humans.</span></span> <span data-ttu-id="4a52c-126">這個建構函式的呼叫端必須確保這個字串已經被當地語系化 (為了目前系統的文化特性)。</span><span class="sxs-lookup"><span data-stu-id="4a52c-126">The caller of this constructor is required to ensure that this string has been localized for the current system culture.</span></span>  
  
 <span data-ttu-id="4a52c-127">下表顯示的執行個體的初始屬性值<xref:System.ArithmeticException>。</span><span class="sxs-lookup"><span data-stu-id="4a52c-127">The following table shows initial property values for an instance of <xref:System.ArithmeticException>.</span></span>  
  
|<span data-ttu-id="4a52c-128">屬性</span><span class="sxs-lookup"><span data-stu-id="4a52c-128">Property</span></span>|<span data-ttu-id="4a52c-129">值</span><span class="sxs-lookup"><span data-stu-id="4a52c-129">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="4a52c-130">Null 參考 (`Nothing` Visual Basic 中)。</span><span class="sxs-lookup"><span data-stu-id="4a52c-130">The null reference (`Nothing` in Visual Basic).</span></span>|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="4a52c-131">錯誤訊息字串。</span><span class="sxs-lookup"><span data-stu-id="4a52c-131">The error message string.</span></span>|  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="protected ArithmeticException (System.Runtime.Serialization.SerializationInfo info, System.Runtime.Serialization.StreamingContext context);" />
      <MemberSignature Language="ILAsm" Value=".method familyhidebysig specialname rtspecialname instance void .ctor(class System.Runtime.Serialization.SerializationInfo info, valuetype System.Runtime.Serialization.StreamingContext context) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.ArithmeticException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)" />
      <MemberSignature Language="VB.NET" Value="Protected Sub New (info As SerializationInfo, context As StreamingContext)" />
      <MemberSignature Language="C++ CLI" Value="protected:&#xA; ArithmeticException(System::Runtime::Serialization::SerializationInfo ^ info, System::Runtime::Serialization::StreamingContext context);" />
      <MemberSignature Language="F#" Value="new ArithmeticException : System.Runtime.Serialization.SerializationInfo * System.Runtime.Serialization.StreamingContext -&gt; ArithmeticException" Usage="new System.ArithmeticException (info, context)" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>1.0.5000.0</AssemblyVersion>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <Attributes>
        <Attribute FrameworkAlternate="netframework-4.0">
          <AttributeName>System.Security.SecuritySafeCritical</AttributeName>
        </Attribute>
      </Attributes>
      <Parameters>
        <Parameter Name="info" Type="System.Runtime.Serialization.SerializationInfo" />
        <Parameter Name="context" Type="System.Runtime.Serialization.StreamingContext" />
      </Parameters>
      <Docs>
        <param name="info"><span data-ttu-id="4a52c-132">存放序列物件資料的物件。</span><span class="sxs-lookup"><span data-stu-id="4a52c-132">The object that holds the serialized object data.</span></span></param>
        <param name="context"><span data-ttu-id="4a52c-133">關於來源或目的端的內容資訊。</span><span class="sxs-lookup"><span data-stu-id="4a52c-133">The contextual information about the source or destination.</span></span></param>
        <summary><span data-ttu-id="4a52c-134">使用序列化資料，初始化 <see cref="T:System.ArithmeticException" /> 類別的新執行個體。</span><span class="sxs-lookup"><span data-stu-id="4a52c-134">Initializes a new instance of the <see cref="T:System.ArithmeticException" /> class with serialized data.</span></span></summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="4a52c-135">進行還原序列化 (Deserialization) 期間會呼叫這個建構函式，以便重新構成經由資料流傳送的例外狀況物件。</span><span class="sxs-lookup"><span data-stu-id="4a52c-135">This constructor is called during deserialization to reconstitute the exception object transmitted over a stream.</span></span> <span data-ttu-id="4a52c-136">如需詳細資訊，請參閱 < [XML 和 SOAP 序列化](~/docs/standard/serialization/xml-and-soap-serialization.md)。</span><span class="sxs-lookup"><span data-stu-id="4a52c-136">For more information, see [XML and SOAP Serialization](~/docs/standard/serialization/xml-and-soap-serialization.md).</span></span>  
  
 ]]></format>
        </remarks>
        <related type="Article" href="~/docs/standard/serialization/xml-and-soap-serialization.md"><span data-ttu-id="4a52c-137">XML 和 SOAP 序列化</span><span class="sxs-lookup"><span data-stu-id="4a52c-137">XML and SOAP Serialization</span></span></related>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public ArithmeticException (string message, Exception innerException);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message, class System.Exception innerException) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.ArithmeticException.#ctor(System.String,System.Exception)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (message As String, innerException As Exception)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; ArithmeticException(System::String ^ message, Exception ^ innerException);" />
      <MemberSignature Language="F#" Value="new ArithmeticException : string * Exception -&gt; ArithmeticException" Usage="new System.ArithmeticException (message, innerException)" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.10.0</AssemblyVersion>
        <AssemblyVersion>4.0.20.0</AssemblyVersion>
        <AssemblyVersion>4.1.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>1.0.5000.0</AssemblyVersion>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="message" Type="System.String" />
        <Parameter Name="innerException" Type="System.Exception" />
      </Parameters>
      <Docs>
        <param name="message"><span data-ttu-id="4a52c-138">解釋例外狀況原因的錯誤訊息。</span><span class="sxs-lookup"><span data-stu-id="4a52c-138">The error message that explains the reason for the exception.</span></span></param>
        <param name="innerException"><span data-ttu-id="4a52c-139">做為目前例外狀況發生原因的例外狀況。</span><span class="sxs-lookup"><span data-stu-id="4a52c-139">The exception that is the cause of the current exception.</span></span> <span data-ttu-id="4a52c-140">如果 <paramref name="innerException" /> 參數不是 null 參考，目前的例外狀況會在處理內部例外的 <see langword="catch" /> 區塊中引發。</span><span class="sxs-lookup"><span data-stu-id="4a52c-140">If the <paramref name="innerException" /> parameter is not a null reference, the current exception is raised in a <see langword="catch" /> block that handles the inner exception.</span></span></param>
        <summary><span data-ttu-id="4a52c-141">使用指定的錯誤訊息以及造成此例外狀況的內部例外狀況的參考，初始化 <see cref="T:System.ArithmeticException" /> 類別的新執行個體。</span><span class="sxs-lookup"><span data-stu-id="4a52c-141">Initializes a new instance of the <see cref="T:System.ArithmeticException" /> class with a specified error message and a reference to the inner exception that is the cause of this exception.</span></span></summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="4a52c-142">被擲回以做為前一個例外狀況直接結果的例外狀況，應該在 <xref:System.Exception.InnerException%2A> 屬性中包含對前一個例外狀況的參考。</span><span class="sxs-lookup"><span data-stu-id="4a52c-142">An exception that is thrown as a direct result of a previous exception should include a reference to the previous exception in the <xref:System.Exception.InnerException%2A> property.</span></span> <span data-ttu-id="4a52c-143"><xref:System.Exception.InnerException%2A> 屬性會傳回與傳入建構函式中相同的值；如果 `Nothing` 屬性不提供內部例外值給建構函式，則傳回 null 參考 (在 Visual Basic 中為 <xref:System.Exception.InnerException%2A>)。</span><span class="sxs-lookup"><span data-stu-id="4a52c-143">The <xref:System.Exception.InnerException%2A> property returns the same value that is passed into the constructor, or a null reference (`Nothing` in Visual Basic) if the <xref:System.Exception.InnerException%2A> property does not supply the inner exception value to the constructor.</span></span>  
  
 <span data-ttu-id="4a52c-144">下表顯示的執行個體的初始屬性值<xref:System.ArithmeticException>。</span><span class="sxs-lookup"><span data-stu-id="4a52c-144">The following table shows initial property values for an instance of <xref:System.ArithmeticException>.</span></span>  
  
|<span data-ttu-id="4a52c-145">屬性</span><span class="sxs-lookup"><span data-stu-id="4a52c-145">Property</span></span>|<span data-ttu-id="4a52c-146">值</span><span class="sxs-lookup"><span data-stu-id="4a52c-146">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="4a52c-147">內部例外狀況參考。</span><span class="sxs-lookup"><span data-stu-id="4a52c-147">The inner exception reference.</span></span>|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="4a52c-148">錯誤訊息字串。</span><span class="sxs-lookup"><span data-stu-id="4a52c-148">The error message string.</span></span>|  
  
 ]]></format>
        </remarks>
        <altmember cref="T:System.Exception" />
        <related type="Article" href="~/docs/standard/exceptions/index.md"><span data-ttu-id="4a52c-149">處理和擲回例外狀況</span><span class="sxs-lookup"><span data-stu-id="4a52c-149">Handling and Throwing Exceptions</span></span></related>
      </Docs>
    </Member>
  </Members>
</Type>